### PR TITLE
Fix - added preview for post editioning and when none answers yet

### PR DESCRIPTION
**Pull request by ChrissP92**
**Fix to** https://github.com/CodersCommunity/forum.pasja-informatyki.local/pull/24

Post preview is now also available when:
- user edits his question, answer or comment
- topic has zero answers

### CODE REVIEW

**Comment by efiku** [comment]
Myślę, że może być okej. 
Można merge + squash .

File: forum/qa-content/qa-page.js [hunk 1]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+				p.innerHTML = singleLineOfCode;*/
+				
+				////console.log('PPP: ', p);
+				textArea.value += singleLineOfCode + '\r\n';
+			});
+			
+			document.body.appendChild(textArea);
+			////console.log('textArea: ', textArea.value);
+			
+			// if anything on the page is selected (a.k.a highlighted) - clear the selection
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 2]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+				/*var p = document.createElement('p');
+				
+				p.innerHTML = singleLineOfCode;*/
+				
+				////console.log('PPP: ', p);
+				textArea.value += singleLineOfCode + '\r\n';
+			});
+			
+			document.body.appendChild(textArea);
+			////console.log('textArea: ', textArea.value);
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 3]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+			textArea.classList.add('content-copy');
+			
+			////console.log('codee: ', code);
+			code.forEach(function(singleLineOfCode)
+			{
+				/*var p = document.createElement('p');
+				
+				p.innerHTML = singleLineOfCode;*/
+				
+				////console.log('PPP: ', p);
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 4]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+			 * However selecting is only possible on HTML elements that are 'inputs', such as <textarea>.
+			 * That's why below code creates <textarea> and insert there code content from block, copy it to clipboard and removes it (so user can't really see temporary <textarea> element blink) after whole process.
+			 */
+			var textArea = document.createElement("textarea");
+			textArea.classList.add('content-copy');
+			
+			////console.log('codee: ', code);
+			code.forEach(function(singleLineOfCode)
+			{
+				/*var p = document.createElement('p');
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 5]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+			/*
+			 *	In order to be able to copy the code inside block into the clipboard - so user can easily paste it wherever he wants - within single button click
+			 * a code must be first selected (or highlighted in human meaning), so JavaScript can copy it.
+			 * However selecting is only possible on HTML elements that are 'inputs', such as <textarea>.
+			 * That's why below code creates <textarea> and insert there code content from block, copy it to clipboard and removes it (so user can't really see temporary <textarea> element blink) after whole process.
+			 */
+			var textArea = document.createElement("textarea");
+			textArea.classList.add('content-copy');
+			
+			////console.log('codee: ', code);
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 6]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+	 *	Testing "copy to clipboard" feature
+	 */
+	function copyToClipboard(ev)
+	{
+		// check whether browser support 'copy' command
+		////if (document.queryCommandSupported('copy'))
+		{
+			console.log('[CTRL + C] So let\'s copy some text!');
+			
+			// prevent page refresh (or something weird) as default button action
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 7]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+	// boolean flag, to check if browser supports 'copy' command
+	var prepareClipboardCopying = document.queryCommandSupported('copy');
+	
+	/*
+	 *	Testing "copy to clipboard" feature
+	 */
+	function copyToClipboard(ev)
+	{
+		// check whether browser support 'copy' command
+		////if (document.queryCommandSupported('copy'))
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 8]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
 	
+	// boolean flag, to check if browser supports 'copy' command
+	var prepareClipboardCopying = document.queryCommandSupported('copy');
+	
+	/*
+	 *	Testing "copy to clipboard" feature
+	 */
+	function copyToClipboard(ev)
+	{
+		// check whether browser support 'copy' command
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 9]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
 	
 	'use strict';
 	
+	// boolean flag, to check if browser supports 'copy' command
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment
variable can be renamed to:
```js
var isCopyCommandSupported = document.queryCommandSupported('copy');

```

File: forum/qa-content/qa-page.js [hunk 10]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+			});
+			
+			document.body.appendChild(textArea);
+			////console.log('textArea: ', textArea.value);
+			
+			// if anything on the page is selected (a.k.a highlighted) - clear the selection
+			if (window.getSelection().rangeCount)
+				window.getSelection().removeAllRanges();
+			
+			/*
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Why you always commit shit like this?
Unused code should be removed!

File: forum/qa-content/qa-page.js [hunk 11]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+				// Create Range Object, so that text content can be selected (a.k.a highlighted)
+				var range = document.createRange();
+				range.selectNode( textArea );
+				window.getSelection().addRange(range);
+			}
+			
+			// copy content that is select inside Document - so that is only textarea
+			document.execCommand('copy');
+			
+			// remove <textarea> from DOM
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 12]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+
+			else */ //if (window.getSelection) 
+			{
+				// Create Range Object, so that text content can be selected (a.k.a highlighted)
+				var range = document.createRange();
+				range.selectNode( textArea );
+				window.getSelection().addRange(range);
+			}
+			
+			// copy content that is select inside Document - so that is only textarea
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment

File: forum/qa-content/qa-page.js [hunk 13]
```diff
@@ -193,12 +193,93 @@ function qa_ajax_error()
...
+			/*if (document.selection) 
+			{
+				var range = document.body.createTextRange();
+				range.moveToElementText(document.getElementById(containerid));
+				range.select();
+			}
+
+			else */ //if (window.getSelection) 
+			{
+				// Create Range Object, so that text content can be selected (a.k.a highlighted)
```
Comment (efiku) [inline forum/qa-content/qa-page.js]: Unnecessary comment.

**Comment by efiku** [comment]
I can't merge right now this.
It contains unnecessary comments in code and unused code.
Seriously, try to learn how to use a debugger and your code will be better. 
When you fix this, please squash your commits and remove label **Work in progress**

**Comment by ChrissP92** [comment]
@efiku
I know there are unnecessary comments in code. However i still improve code and what's more, i am adding few new features. So i think, i will do new PR with all clean and working, because comparing to this PR, there are going to be fixes and new features.

Sorry for messing with that current PR. I just have been pushing new changes to send links for users on IRC, who maybe wanted to test how it's working in reality and propose some changes (not for code review indeed).

**Comment by ChrissP92** [comment]
Fixes and improvements are placed here: https://github.com/CodersCommunity/forum.pasja-informatyki.local/pull/34

**Pull request closed without merging**